bookwyrm/static/css/bookwyrm.css

```diff
@@ -3,6 +3,12 @@ html {
     scroll-padding-top: 20%;
 }
 
+body {
+    min-height: 100vh;
+    display: flex;
+    flex-direction: column;
```

Comment:
what's function are the flex attributes serving here?

Reply:
If I only define a `min-height` at `100vh` ( = 100% of the height), the Body will have the size of the window, but the children will be stacked at the top.

What I want is to have the Navbar to be at the top and the Footer at the bottom, both only taking what space they need but no more. Then, the main Section should be able to take whatever space is between the Navbar and the Footer, and more if needed.

So I set the Body as a flexbox, and I set its direction as a `column`. Now, nothing has changed visually: the three children are still stacked at the start of the column. What I want is for the Section to be able to grow as needed.

By default, flexbox children can not grow. So I added  the Bulma helper class `is-flex-grow-1` to the Section, telling it that whatever space it needs, it can take. 

Here's the **before**, with Firefox FlexBox highlighting:

![Screenshot_2021-04-15 BookWyrm(2)](https://user-images.githubusercontent.com/1781070/114934708-8ca75380-9e3a-11eb-8675-ad0a3defd614.png)

The unused space is the hatched zone.

Here's the **after**:

![Screenshot_2021-04-15 BookWyrm(3)](https://user-images.githubusercontent.com/1781070/114934778-a6e13180-9e3a-11eb-9953-74cebe6dbaca.png)

Now the content section takes up all the available space

Here's the **after**, with a lot of content:

![Screenshot_2021-04-15 BookWyrm(4)](https://user-images.githubusercontent.com/1781070/114934839-be201f00-9e3a-11eb-98e2-a291e232bb31.png)

The content section takes up all the space it needs, and the Navbar and the Footer always have the same height.

Reply:
got it, thank you for the detailed explanation!